modules/apps/portal-settings/portal-settings-web/src/main/java/com/liferay/portal/settings/web/internal/portlet/action/EditCompanyMVCActionCommand.java

```diff
@@ -314,9 +315,33 @@ protected void updateCompany(ActionRequest actionRequest) throws Exception {
 			industry, type, size, languageId, timeZoneId, addresses,
 			emailAddresses, phones, websites, unicodeProperties);
 
+		_updateCompanySitesLocale(companyId, unicodeProperties);
+
 		_portal.resetCDNHosts();
 	}
 
+	private void _updateCompanySitesLocale(
+		long companyId, UnicodeProperties unicodeProperties) {
+
+		List<Group> groups = _groupLocalService.getActiveGroups(
+			companyId, true);
+
+		String newCompanyLanguageIds = unicodeProperties.getProperty(
+			PropsKeys.LOCALES);
+
+		for (Group group : groups) {
+			if (group.getType() == GroupConstants.TYPE_SITE_OPEN) {
+				UnicodeProperties groupTypeSettingsUnicodeProperties =
+					group.getTypeSettingsProperties();
+
+				groupTypeSettingsUnicodeProperties.setProperty(
+					PropsKeys.LOCALES, newCompanyLanguageIds);
```

Comment:
This will update the available languages of the group, regardless of what is currently set.  So if we have a group which only allows for English, but we update the instance settings to allow for English and Spanish, now the group will be updated to allow for English and Spanish.

We need to either check for any removed languages and remove those form the groups as well, or make sure the group languages are a subset of the new langauges.

Also, how does this work when a group inherits it's available languages from instance settings?  We need to make sure we make no changes for those group types, because the languages are pulled from the instance settings, not a static list.

Reply:
Thank you for the detailed suggestion!